lib/ledger/managed-ledgerbcoin.js

```diff
@@ -0,0 +1,313 @@
+/*!
+ * managed-ledgerbcoin.js - Managed Ledger Bcoin wrapper.
+ * Copyright (c) 2019, The Bcoin Developers (MIT License).
+ */
+
+'use strict';
+
+const {enforce} = require('bsert');
+const {Device} = require('../device/device');
+const LedgerBcoin = require('./ledgerbcoin');
+
+/**
+ * Wraps Ledger bcoin with device open/close.
+ * @property {Device} device
+ * @property {LedgerBcoin?} ledger
+ */
+class ManagedLedgerBcoin {
+  /**
+   * @param {Object} options
+   * @param {Device} options.device
+   * @param {bcoin.Network} [options.network = primary]
+   * @param {blgr.Logger} [options.logger = global]
+   */
+
+  constructor(options) {
+    enforce(options && typeof options === 'object', 'options', 'object');
+    enforce(options.device instanceof Device, 'options.device', 'Device');
+
+    this.device = options.device;
+    this.bledger = new LedgerBcoin(options);
+  }
+
+  /**
+   * Create Bcoin object from device.
+   * @param {Device} device
+   * @returns {BcoinDevice}
+   */
+
+  static fromDevice(device) {
+    return new this({ device });
+  }
+
+  /**
+   * Open device if not open.
+   */
+
+  async tryOpen() {
+    if (this.device.opened)
+      return;
+
+    await this.device.open();
+  }
+
+  /**
+   * Close device if not closed.
+   */
+
+  async tryClose() {
+    if (!this.device.opened)
+      return;
+
+    await this.device.close();
+  }
+
+  /**
+   * Get firmware version.
+   * @returns {Object}
+   */
+
+  async getFirmwareVersion() {
+    await this.tryOpen();
+
+    let res;
```

Comment:
Would it be better to initialize `res` to `null` in case the call fails and never returns anything?

Reply:
if `res` was not populated it means something threw an error.
So having it `undefined` here is not a big deal.